### PR TITLE
Clean up watcher_tests.go

### PR DESCRIPTION
**Pull request by tallclair**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Follow up from https://github.com/kubernetes/kubernetes/pull/113696, clean up `watcher_tests.go` to use the `/pods` resource base-path.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/sig api-machinery

### CODE REVIEW

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

File: staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go [hunk 1]
```diff
@@ -37,9 +37,6 @@ import (
 	"k8s.io/apiserver/pkg/storage/value"
 )
 
-// basePath for storage keys used across tests
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/testing/utils.go:40]: [This ](https://github.com/kubernetes/kubernetes/pull/113721) achieves this and a bit more.

So I'm going to close this PR in favor of that one (but thank a lot for the cleanup).

**Pull request closed without merging**